discovery-provider/src/utils/ipfs_lib.py

```diff
@@ -146,3 +146,14 @@ def cat(self, multihash):
         except:
             logger.error(f"IPFSCLIENT | IPFS cat timed out for CID {multihash}")
             raise  # error is of type ipfshttpclient.exceptions.TimeoutError
+
+    def update_peers(self, peerslist):
```

Comment:
@hareeshnagaraj What's the overhead of just calling peer a bunch of times over and over? should we have an in-memory cache to avoid unnecessary IO to the IPFS daemon?

Reply:
after looking into this i think we should actually query the peers list - the implementation appears to try and connect even if you are already peered...the overhead of querying the peers list and cross-referencing should be relatively simple

Reply:
implemented in redis